test/unit/gateways/realex_test.rb

```diff
@@ -51,6 +51,41 @@ def setup
     @amount = 100
   end
 
+  def test_initialize_sets_refund_and_credit_hashes
+    refund_secret = 'refund'
+    rebate_secret = 'rebate'
+
+    options = {
+      login: @login,
+      password: @password,
+      rebate_secret: rebate_secret,
+      refund_secret: refund_secret
+    }
+
+    gateway = RealexGateway.new(options)
+
+    set_options = gateway.instance_variable_get(:@options)
+
+    assert set_options[:refund_hash] == Digest::SHA1.hexdigest(rebate_secret)
```

Comment:
I did some testing of this, and the extra step in line 67 isn't needed since you can actually call `options` directly.

```suggestion
    assert gateway.options[:refund_hash] == Digest::SHA1.hexdigest(rebate_secret)
```

Reply:
Great catch -- I hadn't noticed the accessor here: https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L144.